x-pack/plugins/aiops/public/components/change_point_detection/partitions_selector.tsx

```diff
@@ -0,0 +1,148 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { type FC, useState, useCallback, useMemo } from 'react';
+import { EuiComboBox, EuiFormRow } from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+import { type SearchRequest } from '@elastic/elasticsearch/lib/api/types';
+import { EuiComboBoxOptionOption } from '@elastic/eui/src/components/combo_box/types';
+import useMount from 'react-use/lib/useMount';
+import { debounce } from 'lodash';
+import { useAiopsAppContext } from '../../hooks/use_aiops_app_context';
+import { useCancellableSearch } from '../../hooks/use_cancellable_search';
+import { useDataSource } from '../../hooks/use_data_source';
+
+export interface PartitionsSelectorProps {
+  splitField: string;
+  value: string[];
+  onChange: (update: string[]) => void;
+}
+
+function getQueryPayload(indexPattern: string, fieldName: string, queryString: string = '') {
+  return {
+    params: {
+      index: indexPattern,
+      size: 0,
+      aggs: {
+        aggResults: {
+          filter: {
+            bool: {
+              must: {
+                wildcard: {
+                  [fieldName]: {
+                    value: `*${queryString}*`,
+                  },
+                },
+              },
+            },
+          },
+          aggs: {
+            partitionValues: {
+              terms: {
+                field: fieldName,
+              },
+            },
+          },
+        },
+      },
+    } as SearchRequest,
+  };
+}
+
+interface PartitionsResponse {
+  aggregations: {
+    aggResults: {
+      partitionValues: { buckets: Array<{ key: string }> };
+    };
+  };
+}
+
+export const PartitionsSelector: FC<PartitionsSelectorProps> = ({
+  value,
+  onChange,
+  splitField,
+}) => {
+  const { dataView } = useDataSource();
+  const {
+    notifications: { toasts },
+  } = useAiopsAppContext();
+  const [options, setOptions] = useState<Array<EuiComboBoxOptionOption<string>>>([]);
+  const [isLoading, setIsLoading] = useState(true);
+  const { runRequest, cancelRequest } = useCancellableSearch();
+
+  const fetchResults = useCallback(
+    async (searchValue: string) => {
```

Comment:
Might be missing something but when does the `searchValue` get updated for this?

Reply:
thanks for spotting this @alvarezmelissa87! fixed in https://github.com/elastic/kibana/pull/163694/commits/837b991495a383d4de0fd5becda933cb8d67c321